### PR TITLE
Add missing networking RPC calls.

### PR DESCRIPTION
**Pull request by tnull**
This pull request adds the missing networking RPC calls (cf. [API](https://developer.bitcoin.org/reference/rpc/index.html#network-rpcs)). 

I locally tested the calls in regtest as far as I could, looks like they behave as expected.

### CODE REVIEW

**Review by sgeisler — COMMENTED** [review]
Thanks for the PR! Would be awesome if you could also add the methods to our [integration testsuite](https://github.com/rust-bitcoin/rust-bitcoincore-rpc/blob/master/integration_test/src/main.rs) so we can automatically test compatibility with supported versions of bitcoind.

**Comment by tnull** [comment]
> Thanks for the PR! Would be awesome if you could also add the methods to our [integration testsuite](https://github.com/rust-bitcoin/rust-bitcoincore-rpc/blob/master/integration_test/src/main.rs) so we can automatically test compatibility with supported versions of bitcoind.

I now wrote the integration test as far as I could without spawning another bitcoind instance (which would for example become necessary to in order fully test the functionality of the `add_node` call). 

On the way fixed a small bug with the `list_banned` call which is btw. basically undocumented in the [API description](https://developer.bitcoin.org/reference/rpc/listbanned.html).

File: client/src/client.rs
Comment (sgeisler) [inline client/src/client.rs:942]: It seems odd to duplicate defaults here. Ideally we would use the key-value pairs that are left out if `None` was supplied. I think `bitcoind` supports that (at least `bitcoin-cli` does). What do you think @stevenroose or anyone more familiar with this repo?

**Review by sgeisler — APPROVED** [review]
utACK

**Comment by sgeisler** [comment]
I might hold off merging this till #195 is merged so we have CI again. I'd also like a second review, will beg for that on IRC once CI is running again.

**Comment by 0xB10C** [comment]
Will review once rebased on #195 :) Please ping me if I don't.

**Comment by tnull** [comment]
Finally rebased! 😄  (@0xB10C)

File: client/src/client.rs
Comment (0xB10C) [inline client/src/client.rs:912]: I think this documentation is wrong. `fn add_node()` only adds. `fn remove_node()` and `fn onetry_node()` are missing documentation.

**Review by 0xB10C — APPROVED** [review]
crACK 537fa9e8b9f643825289974e60c6b48566d637ac  modulo missing function documentation and https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/171#discussion_r694582821

File: client/src/client.rs
Comment (0xB10C) [reply client/src/client.rs:965]: I had to look at https://developer.bitcoin.org/reference/rpc/setban.html to understand what you mean. I agree that it's odd to default to this here while Bitcoin Core defaults to these values too. 

I think it's cleaner to remove the `bantime` and `absolute` `Option`s here. Allows the user to explicitly specify what he wants without risking that this implementation or Bitcoin Core's RPC API changes.

If the `Option`s are not dropped, the defaults should be documented.

Comment (tnull) [reply client/src/client.rs]: Alright, I removed the `Option`s in https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/171/commits/8b92fa3f82a300f3291b87a0e079d8266dd94dd2.

File: client/src/client.rs
Comment (tnull) [reply client/src/client.rs]: Thanks, fixed the documentation with https://github.com/rust-bitcoin/rust-bitcoincore-rpc/pull/171/commits/8b92fa3f82a300f3291b87a0e079d8266dd94dd2

File: client/src/client.rs
Comment (0xB10C) [inline client/src/client.rs:964]: nit: 
```suggestion
    /// Attempts to add an IP/Subnet to the banned list.
```

File: client/src/client.rs
Comment (0xB10C) [inline client/src/client.rs:911]:
```suggestion
    /// Attempts to add a node to the addnode list.
```

**Review by 0xB10C — COMMENTED** [review]
not sure about conventions for this repo, but could also squash your commits into one

File: client/src/client.rs
Comment (tnull) [reply client/src/client.rs]: Fixed in https://github.com/rust-bitcoin/rust-bitcoincore-rpc/commit/585dfdd62532d200895744b7f007b901b0bcd43a.

**Comment by tnull** [comment]
Thanks for the review! Squashed now.

**Comment by 0xB10C** [comment]
utACK 585dfdd62532d200895744b7f007b901b0bcd43a

**Comment by RCasatta** [comment]
It looks there are some issues in the CI

**Comment by tnull** [comment]
> > `message: "Bitcoin Core is not connected!"`
> 
> Bitcoin Core needs to be connected to at least one peer when calling `getblocktemplate`.

I see. I should have fixed it now by manually adding the original node again at the beginning of `test_getblocktemplate`. Will squash again as soon as this is confirmed by CI (on my local machine integration tests keep failing with quite different errors, might have a look at that next).

**Comment by tnull** [comment]
I fiddled around bit with re-connecting the peer, but found that it's much simpler and cleaner to move the network tests to the end of the integration tests, so that it doesn't matter if the peer isn't connected anymore afterwards. So it should be fixed now.

**Review by RCasatta — APPROVED** [review]
utACK 150953a1382eb3fa3f39cc5802ddc3bc87938fd1

going to merge tomorrow if there isn't any comment from someone else (feel free to ping me if I don't)

**Pull request merged**
